src/Microsoft.Identity.Client/Instance/AuthorityEndpointResolutionManager.cs

```diff
@@ -0,0 +1,171 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.AppConfig;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Exceptions;
+using Microsoft.Identity.Client.OAuth2;
+
+namespace Microsoft.Identity.Client.Instance
+{
+    internal class AuthorityEndpointResolutionManager : IAuthorityEndpointResolutionManager
+    {
+        private static readonly ConcurrentDictionary<string, AuthorityEndpointCacheEntry> EndpointCacheEntries =
```

Comment:
>private static readonly Co [](start = 8, length = 26)

this one should probably have a comment saying why the static is needed and why it's ok?